modules/terraform-aci-contract/main.tf

```diff
@@ -54,10 +83,74 @@ resource "aci_rest_managed" "vzRsSubjFiltAtt" {
 }
 
 resource "aci_rest_managed" "vzRsSubjGraphAtt" {
-  for_each   = { for subj in var.subjects : subj.name => subj if subj.service_graph != null }
+  for_each   = { for subj in var.subjects : subj.name => subj if subj.service_graph != null && length(subj.filters) != 0 }
   dn         = "${aci_rest_managed.vzSubj[each.key].dn}/rsSubjGraphAtt"
   class_name = "vzRsSubjGraphAtt"
   content = {
     tnVnsAbsGraphName = each.value.service_graph
   }
 }
+
+resource "aci_rest_managed" "vzInTerm" {
+  depends_on = [aci_rest_managed.vzSubj]
+  for_each   = { for subj in var.subjects : subj.name => subj if length(subj.filters) == 0 }
+  dn         = "${aci_rest_managed.vzBrCP.dn}/subj-${each.value.name}/intmnl"
+  class_name = "vzInTerm"
+  content = {
+    prio       = each.value.consumer_to_provider_qos_class
+    targetDscp = each.value.consumer_to_provider_target_dscp
+  }
+}
+
+resource "aci_rest_managed" "vzOutTerm" {
+  depends_on = [aci_rest_managed.vzSubj]
+  for_each   = { for subj in var.subjects : subj.name => subj if length(subj.filters) == 0 }
+  dn         = "${aci_rest_managed.vzBrCP.dn}/subj-${each.value.name}/outtmnl"
```

Comment:
Remove `depends_on` and change `dn` to `${aci_rest_managed.vzSubj.dn}/outtmnl`

Reply:
Can you take a look at this comment again? The for_each cant be run like this.

Reply:
Typo on my behalf. Should be `${aci_rest_managed.vzSubj[each.key].dn}/outtmnl`